edxval/utils.py

```diff
@@ -243,3 +245,40 @@ def validate_generated_images(value, max_items):
         raise ValidationError(u'list must only contain strings.')
 
     return value
+
+
+def generate_file_content_hash(uploaded_file):
+    """
+    Generates SHA256 Content Hash for a File
+
+    Arguments:
+        uploaded_file (UploadedFile): File which will be used for hash generation
+
+    Returns:
+        str sha256 hash
+    """
+    content_hash = hashlib.sha256()
+
+    with closing(uploaded_file.open()) as file_content:
+        for chunk in iter(lambda: file_content.read(65536), b""):
+            if not chunk:
+                break
+            content_hash.update(str(chunk).strip().encode('utf-8'))
+
+    return content_hash.hexdigest()
+
+
+def is_duplicate_file(uploaded_file, file_hash):
+    """
+    Checks file hash to know if its a duplicate file
+
+    Arguments:
+        uploaded_file (UploadedFile): File which will be used for hash generation
+        file_hash (str): SHA256 file hash
+
+    Returns:
+        if file is duplicate (boolean)
+    """
```

Comment:
This should be addressed now. The method takes two files instead, as you requested :+1:

Reply:
@DawoudSheraz I just realized that there won't be any need to change any migrations or model changes related to the transcript content hash, since we're computing the content hash when trying to import the transcript.

Accordingly, I added a commit which removes the field and its migrations completely.

If you'd like me to revert that change, please let me know. 

Computing the content hash every time isn't so efficient, but it's definitely more straight forward from a code point of view.